[Bug] GripNet test error

# 🐛 Bug

test_gripnet failed likely due to PyG upgrade.

## To reproduce

https://github.com/pykale/pykale/runs/4016671465?check_suite_focus=true

**Stack trace/error message**

```
=================================== FAILURES ===================================
_____________________________ test_gripnet_encoder _____________________________

pose_data = <[RuntimeError("The 'data' object was created by an older version of PyG. If this error occurred while loading an alre...e 'processed/' directory in the dataset's root folder and try again.") raised in repr()] Data object at 0x7f1dc66519d0>

    def test_gripnet_encoder(pose_data):
        gg_layers = [32, 16, 16]
        gd_layers = [16, 32]
        dd_layers = [sum(gd_layers), 16]
        gripnet = TypicalGripNetEncoder(
            gg_layers, gd_layers, dd_layers, pose_data.n_d_node, pose_data.n_g_node, pose_data.n_dd_edge_type
        )
>       gripnet(
            pose_data.g_feat,
            pose_data.gg_edge_index,
            pose_data.edge_weight,
            pose_data.gd_edge_index,
            pose_data.train_idx,
            pose_data.train_et,
            pose_data.train_range,
        )

tests/embed/test_gripnet.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/torch/nn/modules/module.py:727: in _call_impl
    result = self.forward(*input, **kwargs)
kale/embed/gripnet.py:238: in forward
    z = self.source_graph(source_x, source_edge_index, edge_weight=source_edge_weight, if_catout=True)
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/torch/nn/modules/module.py:727: in _call_impl
    result = self.forward(*input, **kwargs)
kale/embed/gripnet.py:99: in forward
    else net(x, homo_edge_index, edge_weight)
/usr/share/miniconda/envs/test/lib/python3.8/site-packages/torch/nn/modules/module.py:727: in _call_impl
    result = self.forward(*input, **kwargs)
kale/embed/gcn.py:110: in forward
    edge_index, norm = self.norm(edge_index, x.size(0), edge_weight, self.improved, x.dtype)
kale/embed/gcn.py:83: in norm
    edge_index, edge_weight = add_remaining_self_loops(edge_index, edge_weight, fill_value, num_nodes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

edge_index = tensor([[   3,    5,    6,  ..., 1133,   13,   13],
        [   2,    1,    0,  ..., 1616, 1625, 1634]])
edge_attr = tensor([1., 1., 1.,  ..., 1., 1., 1.]), fill_value = 1
num_nodes = 1635

    def add_remaining_self_loops(
            edge_index: Tensor, edge_attr: OptTensor = None,
            fill_value: Union[float, Tensor, str] = None,
            num_nodes: Optional[int] = None) -> Tuple[Tensor, OptTensor]:
        r"""Adds remaining self-loop :math:`(i,i) \in \mathcal{E}` to every node
        :math:`i \in \mathcal{V}` in the graph given by :attr:`edge_index`.
        In case the graph is weighted or has multi-dimensional edge features
        (:obj:`edge_attr != None`), edge features of non-existing self-loops will
        be added according to :obj:`fill_value`.
    
        Args:
            edge_index (LongTensor): The edge indices.
            edge_attr (Tensor, optional): Edge weights or multi-dimensional edge
                features. (default: :obj:`None`)
            fill_value (float or Tensor or str, optional): The way to generate
                edge features of self-loops (in case :obj:`edge_attr != None`).
                If given as :obj:`float` or :class:`torch.Tensor`, edge features of
                self-loops will be directly given by :obj:`fill_value`.
                If given as :obj:`str`, edge features of self-loops are computed by
                aggregating all features of edges that point to the specific node,
                according to a reduce operation. (:obj:`"add"`, :obj:`"mean"`,
                :obj:`"min"`, :obj:`"max"`, :obj:`"mul"`). (default: :obj:`1.`)
            num_nodes (int, optional): The number of nodes, *i.e.*
                :obj:`max_val + 1` of :attr:`edge_index`. (default: :obj:`None`)
    
        :rtype: (:class:`LongTensor`, :class:`Tensor`)
        """
        N = maybe_num_nodes(edge_index, num_nodes)
        mask = edge_index[0] != edge_index[1]
    
        loop_index = torch.arange(0, N, dtype=torch.long, device=edge_index.device)
        loop_index = loop_index.unsqueeze(0).repeat(2, 1)
    
        if edge_attr is not None:
            if fill_value is None:
                loop_attr = edge_attr.new_full((N, ) + edge_attr.size()[1:], 1.)
    
            elif isinstance(fill_value, float):
                loop_attr = edge_attr.new_full((N, ) + edge_attr.size()[1:],
                                               fill_value)
            elif isinstance(fill_value, Tensor):
                loop_attr = fill_value.to(edge_attr.device, edge_attr.dtype)
                if edge_attr.dim() != loop_attr.dim():
                    loop_attr = loop_attr.unsqueeze(0)
                sizes = [N] + [1] * (loop_attr.dim() - 1)
                loop_attr = loop_attr.repeat(*sizes)
    
            elif isinstance(fill_value, str):
                loop_attr = scatter(edge_attr, edge_index[1], dim=0, dim_size=N,
                                    reduce=fill_value)
            else:
>               raise AttributeError("No valid 'fill_value' provided")
E               AttributeError: No valid 'fill_value' provided

/usr/share/miniconda/envs/test/lib/python3.8/site-packages/torch_geometric/utils/loop.py:220: AttributeError
---------------------------- Captured stderr setup -----------------------------

  0%|          | 0.00/2.52M [00:00<?, ?B/s]
100%|██████████| 2.52M/2.52M [00:00<00:00, 30.1MB/s]
=============================== warnings summary ===============================
examples/bindingdb_deepdta/tutorial.ipynb: 43 warnings
examples/cmri_mpca/tutorial.ipynb: 14 warnings
examples/digits_dann_lightn/tutorial.ipynb: 22 warnings
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/jsonschema/validators.py:190: DeprecationWarning: Passing a schema to Validator.iter_errors is deprecated and will be removed in a future release. Call validator.evolve(schema=new_schema).iter_errors(...) instead.
    warnings.warn(

tests/loaddata/test_image_access.py::test_get_train_test[MNIST]
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/torchvision/datasets/mnist.py:480: UserWarning: The given NumPy array is not writeable, and PyTorch does not support non-writeable tensors. This means you can write to the underlying (supposedly non-writeable) NumPy array using the tensor. You may want to copy the array to protect its data or make it writeable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at  /pytorch/torch/csrc/utils/tensor_numpy.cpp:141.)
    return torch.from_numpy(parsed.astype(m[2], copy=False)).view(*s)

tests/pipeline/test_deep_dti.py::test_deep_data
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:530: LightningDeprecationWarning: `trainer.fit(train_dataloader)` is deprecated in v1.4 and will be removed in v1.6. Use `trainer.fit(train_dataloaders)` instead. HINT: added 's'
    rank_zero_deprecation(

tests/pipeline/test_deep_dti.py::test_deep_data
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pytorch_lightning/trainer/data_loading.py:376: UserWarning: Your val_dataloader has `shuffle=True`, it is best practice to turn this off for val/test/predict dataloaders.
    rank_zero_warn(

tests/pipeline/test_deep_dti.py::test_deep_data
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pytorch_lightning/trainer/data_loading.py:326: UserWarning: The number of training samples (4) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
    rank_zero_warn(

tests/pipeline/test_deep_dti.py::test_deep_data
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:678: LightningDeprecationWarning: `trainer.test(test_dataloaders)` is deprecated in v1.4 and will be removed in v1.6. Use `trainer.test(dataloaders)` instead.
    rank_zero_deprecation(

tests/pipeline/test_domain_adapter.py: 9 warnings
tests/pipeline/test_multi_domain_adapter.py: 4 warnings
tests/pipeline/test_video_domain_adapter.py: 21 warnings
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:216: UserWarning: Please also save or load the state of the optimizer when saving or loading the scheduler.
    warnings.warn(SAVE_STATE_WARNING, UserWarning)

tests/pipeline/test_multi_domain_adapter.py::test_multi_source[1-MFSAN]
tests/pipeline/test_multi_domain_adapter.py::test_multi_source[1-M3SDA]
tests/pipeline/test_multi_domain_adapter.py::test_multi_source[1-DIN]
tests/pipeline/test_multi_domain_adapter.py::test_multi_source[2-MFSAN]
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pytorch_lightning/trainer/data_loading.py:326: UserWarning: The number of training samples (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
    rank_zero_warn(

tests/pipeline/test_video_domain_adapter.py: 21 warnings
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pytorch_lightning/trainer/data_loading.py:326: UserWarning: The number of training samples (6) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
    rank_zero_warn(

tests/prepdata/test_image_transform.py::test_reg
  /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pydicom/filereader.py:354: UserWarning: Expected explicit VR, but found implicit VR - using implicit VR for reading
    warnings.warn(msg + f" - using {found_vr} VR for reading", UserWarning)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

---------- coverage: platform linux, python 3.8.12-final-0 -----------
Name                                    Stmts   Miss  Cover
-----------------------------------------------------------
kale/__init__.py                            1      0   100%
kale/embed/__init__.py                      0      0   100%
kale/embed/attention_cnn.py                40      1    98%
kale/embed/gcn.py                          98     35    64%
kale/embed/gripnet.py                      81     26    68%
kale/embed/image_cnn.py                   203      5    98%
kale/embed/mpca.py                        107     16    85%
kale/embed/positional_encoding.py          20      0   100%
kale/embed/seq_nn.py                       36      0   100%
kale/embed/video_feature_extractor.py      59      2    97%
kale/embed/video_i3d.py                   217     21    90%
kale/embed/video_res3d.py                 191      5    97%
kale/embed/video_se_i3d.py                153      2    99%
kale/embed/video_se_res3d.py              148      2    99%
kale/embed/video_selayer.py               100      3    97%
kale/evaluate/__init__.py                   0      0   100%
kale/evaluate/metrics.py                   12      1    92%
kale/interpret/__init__.py                  0      0   100%
kale/interpret/model_weights.py            15      1    93%
kale/interpret/visualize.py                42      2    95%
kale/loaddata/__init__.py                   0      0   100%
kale/loaddata/dataset_access.py            30      5    83%
kale/loaddata/image_access.py             157      5    97%
kale/loaddata/mnistm.py                    77     13    83%
kale/loaddata/multi_domain.py             299     22    93%
kale/loaddata/sampler.py                  199     29    85%
kale/loaddata/tdc_datasets.py              28      2    93%
kale/loaddata/usps.py                      63      2    97%
kale/loaddata/video_access.py              96      2    98%
kale/loaddata/video_datasets.py            59      8    86%
kale/loaddata/video_multi_domain.py       127     32    75%
kale/loaddata/videos.py                    95     12    87%
kale/pipeline/__init__.py                   0      0   100%
kale/pipeline/deep_dti.py                  60      4    93%
kale/pipeline/domain_adapter.py           523     56    89%
kale/pipeline/mpca_trainer.py              73      0   100%
kale/pipeline/multi_domain_adapter.py     170      6    96%
kale/pipeline/video_domain_adapter.py     392     12    97%
kale/predict/__init__.py                    0      0   100%
kale/predict/class_domain_nets.py         100     10    90%
kale/predict/decode.py                     20      0   100%
kale/predict/isonet.py                    204     51    75%
kale/predict/losses.py                     77      1    99%
kale/prepdata/__init__.py                   0      0   100%
kale/prepdata/chem_transform.py            31      0   100%
kale/prepdata/image_transform.py           80      2    98%
kale/prepdata/tensor_reshape.py            26      0   100%
kale/prepdata/video_transform.py           23      3    87%
kale/utils/__init__.py                      0      0   100%
kale/utils/download.py                     35      0   100%
kale/utils/logger.py                       19      0   100%
kale/utils/print.py                         9      0   100%
kale/utils/seed.py                         12      2    83%
-----------------------------------------------------------
TOTAL                                    4607    401    91%

=========================== short test summary info ============================
FAILED tests/embed/test_gripnet.py::test_gripnet_encoder - AttributeError: No...
===== 1 failed, 221 passed, 2 skipped, 144 warnings in 1525.72s (0:25:25) ======
Error: Process completed with exit code 1.
```

## Expected Behaviour

Test fails.

## Additional context
@pz-white Consider to make the code compatible with newer version of PyG or restrict the version of PyG in installation as in PR #250

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -41,7 +41,7 @@ jobs:
           pip install torch-sparse -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-cluster -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-spline-conv -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
-          pip install torch-geometric --ignore-installed certifi
+          pip install torch-geometric==1.7.0 --ignore-installed certifi
           pip install -e .[dev]
         shell: bash -l {0}
       - name: Run unit tests

